### PR TITLE
feat: Support CSV description file type to have on-chain attributes

### LINKED ISSUE #11282: Support CSV description file type to have on-chain attributes
continue after #11267
On  mass mint page, support CSV description file type to have attributes
image

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Feature
- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #11282

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

image

### CODE REVIEW

**Review by preschian — APPROVED** [review]

File: public/massmint/template.csv [hunk 1]
```diff
@@ -1,5 +1,5 @@
-file,name,description,price
-art1.png,Art #1,Description for the Art #1,1
-art2.png,Art #2,Description for the Art #2,1
-art3.png,Art #3,Description for the Art #3,1
-art4.png,Art #4,Description for the Art #4,1
+file,name,description,attributes,price
+art1.png,Art #1,Description for the Art #1,"color:white;expression:happy",1
+art2.png,Art #2,Description for the Art #2,"color:blue;expression:shy",1
```
Comment (vikiival) [inline public/massmint/template.csv:3]: why does it have quotes? (i mean `"`)
Is there any special reason for that ?

File: composables/massmint/parsers/parseCsv.ts [hunk 2]
```diff
@@ -1,70 +1,69 @@
...
+    for (const [j, header] of csvHeaders.entries()) {
+      const value = values[j]?.trim()
+      if (header === 'price') {
+        entry.price = parseFloat(value) || undefined
+      }
+      else if (header === 'attributes') {
+        if (value) {
+          const attributes: OpenSeaAttribute[] = []
+          const attrs: Record<string, string> = {}
+          value.split(';').forEach((attr) => {
```
Comment (vikiival) [inline composables/massmint/parsers/parseCsv.ts:29]: move to constant on the top of the file `const ATTRIBUTE_SEPARATOR = ';'`

File: composables/massmint/parsers/parseCsv.ts [hunk 3]
```diff
@@ -1,70 +1,69 @@
...
-}
+const DEFAULT_CSV_HEADERS = ['file', 'name', 'description', 'attributes', 'price']
 
-function parseCsvWithHeaders(lines, csvHeaders): Record<string, Entry> {
+function processCsvLines(lines: string[], csvHeaders: string[]): Record<string, Entry> {
   const entries: Record<string, Entry> = {}
 
   for (const line of lines) {
-    const entry = processCsvLine(line.trim(), csvHeaders)
+    const values = line.split(',').map(value => removeQuotes(value.trim()))
```
Comment (vikiival) [inline composables/massmint/parsers/parseCsv.ts:11]: `LINE_SEPARATOR`

**Review by vikiival — COMMENTED** [review]
~

File: public/massmint/template.csv (hunk 1 above)
Comment (Jarsen136) [reply public/massmint/template.csv]: No special reason and I removed the quotes. It works the same

File: composables/massmint/parsers/parseCsv.ts (hunk 2 above)
Comment (Jarsen136) [reply composables/massmint/parsers/parseCsv.ts]: updated

File: composables/massmint/parsers/parseCsv.ts (hunk 3 above)
Comment (Jarsen136) [reply composables/massmint/parsers/parseCsv.ts:14]: updated

**Review by vikiival — APPROVED** [review]

**Pull request merged**